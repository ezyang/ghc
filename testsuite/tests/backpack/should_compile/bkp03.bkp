-- NB: This example doesn't totally work; we overwrote the
-- old signature H with a new one, which is not the best.

package p where
    signature H(T) where
        data T
    module A(T) where
        import H(T)
    module B(T) where
        data T = T
    signature H(T, f) where
        import B(T)
        f :: a -> a
