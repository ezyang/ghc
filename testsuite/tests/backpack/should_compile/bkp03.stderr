[1 of 1] Processing p
/--- Shape for p
provides: B -> p(hole:H):B
               T{T, T}
          A -> p(hole:H):A
               p(hole:H):B.T{p(hole:H):B.T}
requires: H -> p(hole:H):H, p(hole:H):H
               p(hole:H):B.T{p(hole:H):B.T, p(hole:H):B.T}, f
signature hole:H
module p(hole:H):B
signature hole:H
module p(hole:H):A
\---
  [1 of 4] Compiling H[abstract sig]  ( p/H.hsig, nothing )
  [2 of 4] Compiling B                ( p/B.hs, nothing )
  [3 of 4] Compiling H[abstract sig]  ( p/H.hsig, nothing )
  [4 of 4] Compiling A                ( p/A.hs, nothing )
